compute/apiv1/smoke_test.go

```diff
@@ -0,0 +1,257 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// To run these tests, set GCLOUD_TESTS_GOLANG_PROJECT_ID env var to your GCP projectID
+
+package compute
+
+import (
+	"cloud.google.com/go/internal/testutil"
+	"cloud.google.com/go/internal/uid"
+	"context"
+	"fmt"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+	"google.golang.org/protobuf/proto"
+	"testing"
+)
+
+var projectId = testutil.ProjID()
+var defaultZone = "us-central1-a"
+
+func TestCreateGetListInstance(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping smoke test in short mode")
+	}
+	space := uid.NewSpace("gogapic", nil)
+	name := space.New()
+	ctx := context.Background()
+	c, err := NewInstancesRESTClient(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	zonesClient, err := NewZoneOperationsRESTClient(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	diskType := computepb.AttachedDisk_PERSISTENT
+	instance := &computepb.Instance{
+		Name:        &name,
+		Description: proto.String("тест"),
+		MachineType: proto.String(fmt.Sprintf("https://www.googleapis.com/compute/v1/projects/%s/zones/%s/machineTypes/n1-standard-1", projectId, defaultZone)),
+		Disks: []*computepb.AttachedDisk{
+			{
+				AutoDelete: proto.Bool(true),
+				Boot:       proto.Bool(true),
+				Type:       &diskType,
+				InitializeParams: &computepb.AttachedDiskInitializeParams{
+					SourceImage: proto.String("projects/debian-cloud/global/images/family/debian-10"),
+				},
+			},
+		},
+		NetworkInterfaces: []*computepb.NetworkInterface{
+			{
+				AccessConfigs: []*computepb.AccessConfig{
+					{
+						Name: proto.String("default"),
+					},
+				},
+			},
+		},
+	}
+	createRequest := &computepb.InsertInstanceRequest{
+		Project:          projectId,
+		Zone:             defaultZone,
+		InstanceResource: instance,
+	}
+
+	insert, err := c.Insert(ctx, createRequest)
+	if err != nil {
+		t.Error(err)
```

Comment:
nit: Please add a little extra information to error message about what the failure is. Also, should this be a t.Fatal? (The rest of the test would stop running.)